File: Daily Java to TS/AddTwoNumbers.ts
// This program adds two numbers entered by the user
import * as readline from 'readline'; // Import the readline module for user input

class AddTwoNumbers {
    // Main method to execute the program
    static main(): void {
        // Create an interface for reading input from the console
        const rl = readline.createInterface({
            input: process.stdin, // Standard input
            output: process.stdout // Standard output
        });
        // Ask the user for the first number
        rl.question('Enter first number: ', (num1) => {
            // Ask the user for the second number
            rl.question('Enter second number: ', (num2) => {
                // Parse the input strings to integers and add them
                const sum = parseInt(num1) + parseInt(num2);
                // Print the sum to the console
                console.log(`Sum: ${sum}`);
                // Close the readline interface
                rl.close();
            });
        });
    }
}

// Call the main method to run the program
AddTwoNumbers.main();

File: Daily Java to TS/AddTwoNumbers.java
// This program adds two numbers entered by the user
import java.util.Scanner; // Import the Scanner class for user input

public class AddTwoNumbers {
    // Main method to execute the program
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in); // Create a Scanner object for input
        System.out.print("Enter first number: "); // Prompt user for first number
        int num1 = scanner.nextInt(); // Read the first number
        System.out.print("Enter second number: "); // Prompt user for second number
        int num2 = scanner.nextInt(); // Read the second number
        int sum = num1 + num2; // Add the two numbers
        System.out.println("Sum: " + sum); // Print the sum
        scanner.close(); // Close the Scanner
    }
}
